src/NetAnalyzers/CSharp/Microsoft.CodeQuality.Analyzers/Maintainability/CSharpUseCrossPlatformIntrinsicsFixer.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System;
+using System.Composition;
+using System.Diagnostics;
+using Analyzer.Utilities.Lightup;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeQuality.Analyzers.Maintainability;
+
+namespace Microsoft.CodeQuality.CSharp.Analyzers.Maintainability
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp), Shared]
+    public sealed class CSharpUseCrossPlatformIntrinsicsFixer : UseCrossPlatformIntrinsicsFixer
+    {
+        protected override SyntaxNode CreateExclusiveOrExpression(SyntaxNode left, SyntaxNode right)
+            => SyntaxFactory.BinaryExpression(SyntaxKind.ExclusiveOrExpression, (ExpressionSyntax)left, (ExpressionSyntax)right);
+
+        protected override SyntaxNode CreateLeftShiftExpression(SyntaxNode left, SyntaxNode right)
+            => SyntaxFactory.BinaryExpression(SyntaxKind.LeftShiftExpression, (ExpressionSyntax)left, (ExpressionSyntax)right);
+
+        protected override SyntaxNode CreateRightShiftExpression(SyntaxNode left, SyntaxNode right)
+            => SyntaxFactory.BinaryExpression(SyntaxKind.RightShiftExpression, (ExpressionSyntax)left, (ExpressionSyntax)right);
+
+        protected override SyntaxNode? CreateUnsignedRightShiftExpression(SyntaxNode left, SyntaxNode right)
+        {
+            const LanguageVersion CSharp11 = (LanguageVersion)1100;
```

Comment:
Note: i believe @jjonescz is updating the version of Roslyn we reference.  So ideally this can go away soon.

Reply:
Only the version in tests, not the version the analyzers depend on (so they can continue to be used in older projects)

Reply:
fyi @jjonescz @JoeRobich @jaredpar, i believe we removed that in the analyzers/fixers we ported over to Roslyn itself.  The maintenance burden of trying to support N versions backwards is just too high.  I think we decided we'll support up to one behind the last released version.  That way we can remove tons of the hacks for this sort of time at a reasonable cadence as we move roslyn forward.

I'll let jared weigh in on what the full plan is.  NOte: this is all non-blocking for this PR.  We can move foward with thsi appraoch, and hten clean up as appropriate when we make future changes to what these projects reference.

this is needed now more than ever with our lower resourcing for tehse projects. 

Thanks Jan!